src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+export interface Figure {
+  color: Color;
+  shape: Shape;
+}
+
+function validSquare(square: number): number {
+  return Math.trunc(square * 100) / 100;
+}
 
+function negativeCheck(value: number): void {
+  if (value <= 0) {
+    throw new Error('The figure`s values should be more than 0');
+  }
 }
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public firstSide: number,
+    public secondSide: number,
+    public thirdSide: number,
+    public shape = Shape.Triangle,
```

Comment:
you shouldn't set `shape `in the constructor, because it will be always `triangle `for this class

Reply:
fix it everywhere